### ISSUE TITLE
Navigating to the last Grid page when grouping is enabled results in an error

### ISSUE THREAD

**Issue by martintabakov** [telerik/kendo-ui-core#7589]
### Bug report

An error is thrown in the browser console when you navigate to the last Grid page when grouping is enabled.

**Regression with 2022.1.301**

### Reproduction of the problem

1. Open this Dojo example - https://dojo.telerik.com/aYInodep/3
2. Expand the Canada group.
3. Navigate to the last page

### Current behavior

An error is thrown on the browser console.

### Expected/desired behavior

No errors should be thrown in the browser console

### Environment

* **Kendo UI version:** 2023.3.1010 
* **Browser:** [all]

**Comment by gyankov** [comment]
After this fix the skip and take parameters are mandatory, thus when server operations are disabled, no data is returned.

**Comment by martintabakov** [comment]
Reopening this as the issue still occurs with the latest version - https://dojo.telerik.com/aYInodep/18
Expand the Canada group and go to page 3 -> an error is thrown.

**Comment by ag-petrov** [comment]
We did look into this and corrections were made but I need to clarify a bit on this bug.

When the initial fix was applied it did have a server part included in our **ToDataSourceResult** methods for the MVC/Core products. With the fix **skip** and **take** parameters were used instead of **page** and **pageSize**. The changes were included in the products but were not reflected by our group paging end-point used in the demos(https://demos.telerik.com/aspnet-core/service/api/customers) and in the dojo linked for reproduction. We are in the process of re-hosting an updated version of it which will function correctly. 

With the re-hosted service an error will not occur but while testing I did found a separate problem which is now being tested as changes are implemented. A workaround for the issue is provided in this dojo - https://dojo.telerik.com/@agpetrov/aZAhoyIK . 

NOTE that the dojo will function correctly after we re-hoist the service
NOTE that developers that are using the latest version of the ToDataSourceResult and the override from the dojo should not experience any issues.
